### PR TITLE
[PRM-455] - Added appeal against obligation variation on file upload page

### PR DESCRIPTION
**Pull request by LewisMitchell**

### CODE REVIEW

**Comment by LewisMitchell** [comment]
upload-page

**Review by ruthdas03 — COMMENTED** [review]
Looks good, just a preferential change if you'd like to consider.

File: app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html [hunk 1]
```diff
@@ -9,8 +9,15 @@
 
 @(form: Form[_], postAction: Call)(implicit request: Request[_], messages: Messages, appConfig: AppConfig)
 
-@lppOrLspDynamicText = @{
-    if(request.session.get(SessionKeys.appealType).contains(PenaltyTypeEnum.Late_Payment.toString)) "otherReason.uploadEvidence.lpp.p2" else "otherReason.uploadEvidence.p2"
+@lppOrLspOrObligationDynamicText = @{
```
[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html:12]: Yeah - will take a look, thanks.

[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html:12]: It seems that pattern matching on enums when using toString is not possible (https://stackoverflow.com/a/35487238). As the session keys can only be String's, the toString method has to be called. 😢

[the remark this answers is not in the document]
Comment (ruthdas03) [reply app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html:12]: Would this help, here we are using pattern matching with boolean 
val isLPP = request.session.get(SessionKeys.appealType).contains(PenaltyTypeEnum.Late_Payment.toString)
val isLSP = request.session.get(SessionKeys.appealType).contains(PenaltyTypeEnum.Late_Submission.toString)
val isOA = request.session.get(SessionKeys.isObligationAppeal).isDefined

(isLPP,isLSP, isOA) match {
case (isLPP, isOA, _) => ""
case (isLPP, _, _) => "otherReason.uploadEvidence.lpp.p2"
case (isLSP, isOA, _) => "otherReason.uploadEvidence.appealAgainstObligation.p2"
case (_,_,_) => "otherReason.uploadEvidence.p2"
}

[the remark this answers is not in the document]
Comment (LewisMitchell) [reply app/views/reasonableExcuseJourneys/other/UploadEvidencePage.scala.html:12]: Yeah I can understand how that may work - however I feel by adding this its adding complexity as we are introducing new vals which should really be taken out of the view and put in the controller. I believe the view should remain quite simple and I think the if-else achieves that

**Review by ruthdas03 — APPROVED** [review]

**Pull request merged**